lib/experimental/class-wp-webfonts-provider-local.php

```diff
@@ -96,6 +123,12 @@ public function get_css() {
 			$css .= '@font-face{' . $this->build_font_face_css( $webfont ) . '}';
 		}
 
+		$css = sprintf(
+			"<style id='wp-webfonts-local'%s>\n%s\n</style>\n",
+			$this->type_attr,
+			$css
+		);
```

Comment:
I thought about that. But each Provider will have its own HTML tag needs. For example, a font foundry provider may choose to build a preconnect `<link>` instead of dynamically generating the `@font-face` styles.

Reply:
Hmmm, for resource hints like `preconnect`, providers should be able to use [`wp_resource_hints`](https://developer.wordpress.org/reference/functions/wp_resource_hints/)
Though in practice, I expect all 3rd-party providers will extend the Local provider and download font-files to be hosted locally because of all the court decisions etc lately regarding font-services & GDPR

If we print the `<style>` element here, then perhaps we should rename the function? If it prints stuff, then `the_style` would make more sense than `get_css`

Reply:
To keep the provider's `get_css()` the same, https://github.com/WordPress/gutenberg/pull/43492/commits/dec6a3e792c7a9a445d6af55f8671f5dd0015fd5 and https://github.com/WordPress/gutenberg/pull/43492/commits/d4c01b39ab1feb8ccf7ca3d2f1a63c9cd0ed196f do the following:

* Removes the `<style>` element from `WP_Webfonts_Provider_Local::get_css()`.
* Adds `WP_Webfonts_Provider::get_style_element()` which gets the `<style>` element.
* `WP_Webfonts_Provider::print_styles()` then wraps the CSS from `get_css()` with the `<style>` element.

If a Provider wants something different from `<style>` or its implementation, their Provider can overload the `WP_Webfonts_Provider::get_style_element()` to return what their needs require.

@aristath What do you think?

Reply:
> Though in practice, I expect all 3rd-party providers will extend the Local provider and download font-files to be hosted locally because of all the court decisions etc lately regarding font-services & GDPR

While extenders have that option, some will extend from the abstract class. For example, Jetpack extends `WP_Webfonts_Provider` for its Google Font Provider https://github.com/Automattic/jetpack/blob/b313bd253afdf919adf7a71a866d0ab85b3a7bda/projects/packages/google-fonts-provider/src/class-google-fonts-provider.php#L18.

Reply:
Makes sense 👍